src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,218 @@
+            return String.format("[D] %s %s (by: %s)", Ui.checkbox(this.getIsDone()), this.getDescription(), this.deadline);
+        }
+    }
+
+    /**
+     * Class for a singular Event.
+     */
+    class Event extends Task {
+        String time;
+
+        public Event(String description, String time) {
+            super(description);
+            this.time = time;
+        }
+
+        @Override
+        public String getSavedString() {
+            return String.format("E | %s | %s | %s\n",this.getIsDone(), this.getDescription(), this.time);
+        }
+
+        @Override
+        public String toString() {
+            return String.format("[E] %s %s (at: %s)", Ui.checkbox(this.getIsDone()), this.getDescription(), this.time);
+        }
+    }
+
+    int pointer = 0;
+    protected ArrayList<Task> list;
+    /**
+     * Constructor for TaskList.
+     */
+    public TaskList() {
+        this.list = new ArrayList<>(100);
+    }
+
+    /**
+     * Constructor for TaskList.
+     * @param taskList - tasklist to copy data from
+     */
+    public TaskList(TaskList taskList) {
+        this.list = taskList.list;
+    }
+
+    /**
+     * Adds a task to the list.
+     * @param description - description of the task
+     */
+    public Task addTask(String description) {
+        Task val = new Task(description);
+        this.list.add(val);
+        pointer++;
+        return val;
+    }
+
+    /**
+     * Adds a deadline to the list.
+     * @param desc - description of the task
+     * @param deadline - the date of the deadline
+     */
+    public Deadline addDeadline(String desc, String deadline) {
+        Deadline val = new Deadline(desc, deadline);
+        this.list.add(val);
+        pointer++;
+        return val;
+    }
+
+    /**
+     * Adds a deadline to the list using a date.
+     * @param desc - description of the task
+     * @param date - the date of the deadline in LocalDate format
+     */
+    public Deadline addDeadline(String desc, LocalDate date) {
+        Deadline val = new Deadline(desc, date);
+        this.list.add(val);
+        pointer++;
+        return val;
+    }
+
+    /**
+     * Adds an event to the list.
+     * @param desc - description of the task
+     * @param time - the time of the event
+     */
+    public Event addEvent(String desc, String time) {
+        Event val = new Event(desc, time);
+        this.list.add(val);
+        pointer++;
+        return val;
+    }
+
+    public TaskList find(String keyword) {
+        TaskList result = new TaskList();
+        for (Task task : this.list) {
+            if (task.getDescription().contains(keyword)) {
+                result.list.add(task);
+            }
+        }
+        return result;
+    }
+
+    public int getLength() {
+        return this.list.size();
+    }
+
+    public Task get(int index) {
+        return this.list.get(index);
+    }
+
+    public void delete(int index) {
+        this.list.remove(index);
+    }

```

Comment:
Maybe these could be renamed to getTask and deleteTask just to make the code clearer!